### DISCUSSION TITLE
More?

### DISCUSSION THREAD

**Discussion by MarkMoretto** [rsuite/rsuite#2014] [General]
Can y'all add more to this documentation?

Why, for example, if I have an InputPicker and formRef, does the InputPicker value not get populated when picked?

If I have an InputPicker on a form, as one might logically do, what is the secret method to get the value from the picker into the form key-value map?

**Comment by MarkMoretto** [comment]
Like, at the bottom of [this page](https://rsuitejs.com/components/date-picker/#ranges), what, exactly, is `dateFns.addDays()`?  Some mysterious, "you-should-already-know-it" method that we're supposed to import at some point?

Is [this](https://date-fns.org/v1.29.0/docs/addDays) this package?  Can't import it, sorry.  Maybe it''s a dependency we're just supposed to have for some reason.

Has anyone tried typing in the search box?  Does it work?  What are you allowed to search?  Should we use lowercase only as to not disrupt the flow?

 How did this package become popular enough to remain relevant after a few days?  It seems to make things harder vs better and developers shouldn't induce self flagellation without a good cause.

**Comment by MarkMoretto** [comment]
God, I hope to never use this package again once I'm done with the silly project I'm working on, which utilizes this library, for some silly reason.  Once that's over...peace!  I'll recommend an immediate change from this package to anything else,

**Comment by MarkMoretto** [comment]
What is this even?  Where did the value disappear to?  THe date one works fine.  `SelectPicker` all of a sudden special?

![image](https://user-images.githubusercontent.com/25012707/137643050-e75cd62c-d701-4f59-a0ed-0373eff4ac47.png)

**Comment by MarkMoretto** [comment]
Here's the form inputs.  Pretty basic:

![image](https://user-images.githubusercontent.com/25012707/137643070-418329c0-2e69-4eb2-b406-ded711106a0b.png)

**Comment by MarkMoretto** [comment]
Data goes into a form for what?  Looks and personality?

Maybe, just maybe, that form was created for some reason.  What that reason is, we'll never know.  It's a mystery to mankind.

**Comment by MarkMoretto** [comment]
Does [this](https://github.com/rsuite/rsuite/blob/main/docs/pages/components/select-picker/fragments/async.md) example actually work and if so how?

There's a function that updates state with...some sort of mysterious `data` object?

**Comment by SevenOutman** [comment]
Hi Mark,

Let me answer some of your questions first so that you can manage to make your app work.

> if I have an InputPicker and formRef, does the InputPicker value not get populated when picked?
>
> If I have an InputPicker on a form, as one might logically do, what is the secret method to get the value from the picker into the form key-value map?

RSuite provides a `<Form>` component to handle the cases when  you need to composite multiple inputs as a form. It provides a [controlled `value` prop](https://reactjs.org/docs/forms.html) that holds the key-value pairs, and updates it via `onChange` callback. Here's an example of [using input components with `<Form>` via `<Form.Control>`](https://rsuitejs.com/components/form/#default).

> What is this even? Where did the value disappear to? THe date one works fine. SelectPicker all of a sudden special?

The `<Form>` component in conjunction with input component does not guarantee a equivalent `<form>` element with populated form values so far, but it's indeed what we care and are working to improve as part of semantic&a11y improvement.

> Like, at the bottom of this page, what, exactly, is dateFns.addDays()? Some mysterious, "you-should-already-know-it" method that we're supposed to import at some point?

It's from `date-fns` package as you've already figured out, which is a dependency of RSuite itself since v4. You don't have to import it if you don't need it, the example code block is only to tell what the default values of `<DateRangePicker>` are. But yes we still need to make it more clear as it turned out to be not so obvious as imagined what a semantic API (or just pseudo code) going `addDays(today, 1)` will output without knowing what package it's from.

> Does this example actually work and if so how?
>
> There's a function that updates state with...some sort of mysterious data object?

The `data` object is injected from outside of this example code block as the comment says. If you need to know its structure you can follow the link in the comment.

> Can y'all add more to this documentation?

Yes for sure. 

---

We want to make RSuite better for its users and we're working on it. There can be things we miss and need to be reminded by users. Any positive effort or ideas are welcome.

Feel free to ask me if you have further questions using RSuite. We also recommend using [GitHub Discussions](https://github.com/rsuite/rsuite/discussions) for asking questions instead of Issues where you're supposed to follow the issue template and provide useful information to help us reproduce and locate a functionality failure.
